spec/requests/landing_pages_spec.rb

```diff
@@ -0,0 +1,245 @@
+require "spec_helper"
+
+def create_version(community_id, structure, version_number)
+  s = structure.deep_dup
+  s["page"]["title"] = {value: "Title version #{version_number}"}
+  FactoryGirl.create(
+    :landing_page_version,
+    community_id: community_id,
+    version: version_number,
+    content: JSON.generate(s)
+  )
+end
+
+# Populate category ids in all category sections
+def populate_categories(structure, category_ids)
+  struct = structure.deep_dup
+  struct["sections"].each do |s|
+    next unless s["kind"] == "categories"
+
+    s["categories"].each_with_index do |c, i|
+      c["category"]["id"] = category_ids[i]
+    end
+  end
+  struct
+end
+
+# Populate listing ids in all listing sections
+def populate_listings(structure, listing_ids)
+  struct = structure.deep_dup
+  struct["sections"].each do |s|
+    next unless s["kind"] == "listings"
+
+    s["listings"].each_with_index do |l, i|
+      l["listing"]["id"] = listing_ids[i]
+    end
+  end
+  struct
+end
+
+def release_version(community_id, version_number)
+  lp = LandingPage.where(community_id: community_id).first
+  lp.enabled = 1
+  lp.released_version = version_number
+
+  lp_version = LandingPageVersion.where(community_id: community_id, version: version_number).first
+  lp_version.released = Time.now()
+
+  lp.save!
+  lp_version.save!
+end
+
+def expect_string(url, string)
+  get url
+
+  expect(response.status).to eq(200)
+  expect(response.body).to match(/#{string}/)
+end
+
+def expect_controller(url, controller_name, action_name)
+  get url
+
+  expect(controller.controller_name).to eq(controller_name)
+  expect(controller.action_name).to eq(action_name)
+end
+
+describe "Landing page", type: :request do
+
+  let!(:domain) { "market.custom.org" }
+
+  before do
+    @community = FactoryGirl.create(:community, :domain => domain, use_domain: true)
+    @community.reload
+
+    10.times do
+      FactoryGirl.create(:category, community_id: @community.id)
+    end
+
+    10.times do
+      FactoryGirl.create(:listing, community_id: @community.id)
+    end
+
+    sample_structure = JSON.parse(CustomLandingPage::ExampleData::TEMPLATE_STR)
+    sample_structure = populate_categories(sample_structure, @community.categories.map(&:id))
+    sample_structure = populate_listings(sample_structure, @community.listings.map(&:id))
+
+    create_version(@community.id, sample_structure, 1)
+    create_version(@community.id, sample_structure, 2)
+
+    FactoryGirl.create(
+      :landing_page,
+      community_id: @community.id,
+      enabled: 0,
+      released_version: nil
+    )
+  end
+
+  context "when not released" do
+    it "landing page is disabled" do
+      expect(::CustomLandingPage::LandingPageStore.enabled?(@community.id)).to eq false
```

Comment:
Hhmh... Somehow I feel that this test could (and maybe should) be removed. It's testing the internals of the application and the HTTP interface. What we really want to test is that the routing is correct and that's in the next two tests. What if we in the future change the mechanism how we define if landing page is in use or not? The routing still works but this test may break even though the app is working ok.